packages/babel-register/src/cache.js

```diff
@@ -32,27 +38,66 @@ export function save() {
     }
   }
 
-  mkdirpSync(path.dirname(FILENAME));
-  fs.writeFileSync(FILENAME, serialised);
+  try {
+    mkdirpSync(path.dirname(FILENAME));
+    fs.writeFileSync(FILENAME, serialised);
+  } catch (e) {
+    switch (e.code) {
+      case "EACCES":
+      case "EPERM":
+        console.warn(
+          `Babel could not write cache to file: ${FILENAME} 
+due to a permission issue. Cache is disabled.`,
+        );
+        cacheDisabled = true;
+        break;
+      case "EROFS":
```

Comment:
Disclaimer: I have not tested on a readonly filesystem, this error code is copied from `man 2 open`
> [EROFS]            The named file resides on a read-only file system, and the file is to be modified.